src/overlap_pmmg.c

```diff
@@ -0,0 +1,736 @@
+                   lsPBDY_ToRecv,npPBDY_out2in,MPI_DOUBLE,color_out,MPI_OVERLAP_TAG+LOCAL_MPI_TAG,
+                   comm,&status),return 0 );
+    LOCAL_MPI_TAG++;
+
+    /** STEP 7 - Fill the overlap hash tables with interface points **/
+    for (i=0; i < nitem_ext; i++) {
+      ip_out = pointIdxInterface_ToRecv[i]; // Point index on Pcolor_out (the other partition)
+      ip_in  = pointIdxInterface_ToSend[i]; // Point index on Pcolor_in  (this partition)
+      hash_out2in[ip_out] = ip_in;  // From index on Pcolor_out, I found index on Pcolor_in
+      hash_in2out[ip_in]  = ip_out; // From index on Pcolor_in,  I found index on Pcolor_out
+    }
+
+    /** STEP 8 - Add nodes from Pcolor_out having MG_PARBDY tag to:
+     *     a) the local mesh on Pcolor_in
+     *     b) the overlap hash tables (hash_out2in and hash_in2out)
+     *   Create variable dataPBDY_AlreadyAdded storing:
+     *     (1) color_out: neighbouring partition
+     *     (2) color_ter: tierce partition w/ which MG_PARBDY nodes is shared
+     *     (3) k: position of node ip in external communicator Pcolor_out-Pcolor_ter
+     *     (4) ip_in: local index of node on Pcolor_in
+     *     (5) ip_out: local index of node on Pcolor_out **/
+    mesh->xpmax = MG_MAX( (long long)(1.5*mesh->xp),mesh->npmax);
+
+    /* Loop over the points having MG_PARBDY tag */
+    for (i=0; i < npPBDY_out2in; i++) {
+      ip_out   = pointIdxPBDY_ToRecv[i];
+      coord[0] = pointCoordPBDY_ToRecv[3*i];
+      coord[1] = pointCoordPBDY_ToRecv[3*i+1];
+      coord[2] = pointCoordPBDY_ToRecv[3*i+2];
+      tag      = pointTagPBDY_ToRecv[i]|MG_OVERLAP; // Add the tag MG_OVERLAP to this point
+      ref      = pointRefPBDY_ToRecv[i];
+      ls_val   = lsPBDY_ToRecv[i];
+
+      /* Loop over the data allowing to identify these points */
+      for (j=0; j < ndataPBDY_out2in; j++) {
+        duplicated_point = 0;
+
+        color_ter = dataPBDY_ToRecv[3*j];
+        k         = dataPBDY_ToRecv[3*j+1]; // Position in external communicator Pcolor_out-Pcolor_ter
+        ip        = dataPBDY_ToRecv[3*j+2]; // Index of point on Pcolor_out
+
+        if ( !(ip == ip_out) ) continue;
+
+        /* Search into dataPBDY_AlreadyAdded to seen if it exists already */
+        for (r=0; r < ndataPBDY_added+1; r++) {
+          if ( (color_out == dataPBDY_AlreadyAdded[5*r+1]) && (color_ter == dataPBDY_AlreadyAdded[5*r])) {
+            if ( k == dataPBDY_AlreadyAdded[5*r+2] ) {
+              ip_in = dataPBDY_AlreadyAdded[5*r+3];
+              duplicated_point =1;
+              break;
+            }
+          }
+          if ( (color_out == dataPBDY_AlreadyAdded[5*r]) && (ip_out == dataPBDY_AlreadyAdded[5*r+4])) {
+              ip_in = dataPBDY_AlreadyAdded[5*r+3];
+              duplicated_point =1;
+              break;
+          }
+        }
+
+        /* If this point has not been added yet, then add it to the mesh */
+        if (duplicated_point==0) {
+          if ( color_ter != color_in ) {
```

Comment:
Is it possible that `color_ter == color_in` ?

Reply:
No it is not. I don't understand why I added this condition. I have removed it. Thanks.